server/polar/integrations/stripe/service.py

```diff
@@ -468,11 +469,44 @@ async def update_subscription_price(
                 raise MissingPaymentMethod(id)
             raise
 
-    async def cancel_subscription(self, id: str) -> stripe_lib.Subscription:
-        return await stripe_lib.Subscription.modify_async(
-            id,
-            cancel_at_period_end=True,
-        )
+    async def cancel_subscription(
+        self,
+        id: str,
+        customer_reason: (
+            Literal[
+                "customer_service",
+                "low_quality",
+                "missing_features",
+                "other",
+                "switched_service",
+                "too_complex",
+                "too_expensive",
+                "unused",
+            ]
+            | None
+        ) = None,
+        customer_comment: str | None = None,
+        immediately: bool = False,
+    ) -> stripe_lib.Subscription:
+        details: stripe_lib.Subscription.ModifyParamsCancellationDetails = {}
+        if customer_reason:
+            details["feedback"] = customer_reason
+
+        if customer_comment:
+            details["comment"] = customer_comment
+
+        params: stripe_lib.Subscription.ModifyParams = {
+            "cancellation_details": details,
+        }
+        if immediately:
+            # Need to be an epoch in the future. However, we want the `ended_at`
+            # to ideally have passed upon receiving the webhook. So hoping
+            # one second is a decent balance.
+            params["cancel_at"] = int(time.time()) + 1
```

Comment:
Couldn't we use the cancel API to achieve that?

https://docs.stripe.com/api/subscriptions/cancel

Reply:
True 🤦🏼‍♂️ moment. Can't believe I didn't think of that... Fixing.